plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -571,6 +675,12 @@ fu_plugin_thunderbolt_validate_image (GBytes  *controller_fw,
 			return VALIDATION_FAILED;
 	}
 
+	if (!compare_pd_existence (hw_info->id, &controller, &image, error)) {
+		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
```

Comment:
I think there is still a possibility to overwrite an error here. I think you should move this error set in function for last false condition.

Reply:
Yup, either error_local if you need to change the error code, or just use g_prefix_error if not.

Reply:
That looks fine.